portfolio/src/main/webapp/script.js

```diff
@@ -32,53 +30,27 @@ function addRandomFact() {
 /**
  * Uses fetch() to obtain data from DataServlet.java
  */
- function obtain_fetch_data() {
-     fetch("/data")
-     .then(response => response.text())
-     .then(text => {
-         document.getElementById("fetch-div").innerText = text;
-     })
-     .catch(() => {
-         document.getElementById("fetch-div").innerText = "Error!";
-     })
- }
-
-/**
- * Uses fetch() to obtain json data from DataServlet.java
- */
- function obtain_json_data() {
-    fetch("/data")
-        .then(response => response.json())
-        .then(messages => {
-         console.log(messages)
-        //  const json_div = document.getElementById('json-div');
-        //  json_div.innerHTML = "";
-        //  p = document.createElement("p");
-        //  p.innerText = "Message 1: " + messages[0];
-        //  json_div.appendChild(p);
-        //  p = document.createElement("p");
-        //  p.innerText = "Message 2: " + messages[1];
-        //  json_div.appendChild(p);
-        //  p = document.createElement("p");
-        //  p.innerText = "Message 3: " + messages[2];
-        //  json_div.appendChild(p);
-        })
-        .catch((test) => {
-            document.getElementById("json-div").innerText = "JSON Response Error";
-            console.log(test);
-        })
-        .catch(() => {
-            document.getElementById("json-div").innerText = "JSON Text() Error";
-        });
- }
-
- function obtain_comment() {
-     fetch("/data")
-     .then(response => response.json())
-     .then(comment => {
-         p = document.createElement("p");
-         p.innerText = "Test: " + comment;
-         comment-div.appendChild(p);
-     })
- }
+function obtainFetchData() {
+  document.getElementById('fetch-div').innerText = '';
+  fetch('/data')
+      .then(response => response.json())
+      .then(text_data => {
+        for (var i = 0; i < text_data.length; i++) {
+          p = document.createElement('p');
+          p.innerText =
+              text_data[i].content + '\n' + text_data[i].score_message;
+          document.getElementById('fetch-div').appendChild(p);
+        }
+      })
+      .catch(() => {
```

Comment:
Nit: You can specifically try to log out the error here to console for easier debugging instead of showing an user-facing message. Example:
.catch(error => {
  console.error('Error:', error);
});